<launch>

  <arg name="UAV_NAME" default="$(optenv UAV_NAME uav)" />
  <arg name="name" default="" />
  <arg name="portname" default="/dev/arduino" />
  <arg name="profiler" default="$(optenv PROFILER false)" />

  <arg name="swap_garmins" default="$(optenv SWAP_GARMINS false)" />

  <group ns="$(arg UAV_NAME)">

    <!-- launch the node -->
    <node name="serial$(arg name)" pkg="mrs_serial" type="baca_protocol" output="screen" respawn="true">

      <param name="uav_name" type="string" value="$(arg UAV_NAME)"/>

      <rosparam file="$(find mrs_serial)/config/mrs_serial.yaml" />

      <param name="enable_profiler" type="bool" value="$(arg profiler)" />
      <param name="portname" value="$(arg portname)"/>
      <param name="use_timeout" value="true"/>

      <param name="swap_garmins" value="$(arg swap_garmins)"/>

      <!-- Publishers -->
      <remap from="~range" to="/$(arg UAV_NAME)/garmin/range" />
      <remap from="~range_up" to="/$(arg UAV_NAME)/garmin/range_up" />
      <remap from="~profiler" to="profiler" />
      <remap from="~baca_protocol_out" to="~received_message" />

        <!--ServicesIn-->
      <remap from="~netgun_fire" to="netgun_fire" />
      <remap from="~netgun_arm" to="netgun_arm" />
      <remap from="~netgun_safe" to="netgun_safe" />
      <remap from="~gimbal_pitch" to="gimbal/set_pitch" />

        <!-- Subscribers -->
      <remap from="~baca_protocol_in" to="~send_message" />

    </node>

  </group>

</launch>
